### PR TITLE
fixed couple of issues in task service

### PR DESCRIPTION
**Pull request by llali**

### CODE REVIEW

File: test/Microsoft.SqlTools.ServiceLayer.UnitTests/TaskServices/TaskServiceTests.cs [hunk 1]
```diff
@@ -57,7 +57,7 @@ public void NewTaskShouldSendNotification()
             });
            
             serviceHostMock.Verify(x => x.SendEvent(TaskCreatedNotification.Type,
-                It.Is<TaskInfo>(t => t.TaskId == sqlTask.TaskId.ToString())), Times.Once());
+                It.Is<TaskInfo>(t => t.TaskId == sqlTask.TaskId.ToString() && t.ProviderName == "MSSQL")), Times.Once());
```
Comment (hyoshis) [inline test/Microsoft.SqlTools.ServiceLayer.UnitTests/TaskServices/TaskServiceTests.cs]: why is this necessary?

Comment (llali) [inline test/Microsoft.SqlTools.ServiceLayer.UnitTests/TaskServices/TaskServiceTests.cs]: checking that provider name is mssql? It should always be mssql so just checking that
